# **Phase 2:**

# Read Dataset:

In [2]:
import pandas as pd
import numpy as np

df = pd.read_csv("Dataset/Raw_dataset.csv")

# Attribute Dscription:


In [3]:
data =  data = {
    "Attribute Name": [
        "Age", "Gender", "Ethnicity", "SocioeconomicStatus", "EducationLevel", "BMI", "Smoking",
        "AlcoholConsumption", "PhysicalActivity", "DietQuality", "SleepQuality", "FamilyHistoryDiabetes",
        "GestationalDiabetes","PolycysticOvarySyndrome", "PreviousPreDiabetes", "Hypertension",
        "SystolicBP","DiastolicBP", "FastingBloodSugar", "HbA1c", "SerumCreatinine", "BUNLevels",
        "CholesterolTotal","CholesterolLDL", "CholesterolHDL", "CholesterolTriglycerides",
        "AntihypertensiveMedications", "Statins", "AntidiabeticMedications","FrequentUrination",
        "ExcessiveThirst", "UnexplainedWeightLoss", "FatigueLevels", "BlurredVision", "SlowHealingSores",
        "TinglingHandsFeet", "QualityOfLifeScore","HeavyMetalsExposure", "OccupationalExposureChemicals",
        "WaterQuality", "MedicalCheckupsFrequency", "MedicationAdherence", "HealthLiteracy","Diagnosis"
    ],
    "Description": [
        "The age of the patients", "Gender of the patients", "The ethnicity of the patients", "The socioeconomic status of the patients",
        "The education level of the patients", "Body Mass Index of the patients", "Smoking status", "Weekly alcohol consumption in units",
        "Weekly physical activity in hours", "Diet quality score","Sleep quality score", "Family history of diabetes", "History of gestational diabetes",
        "Presence of polycystic ovary syndrome", "History of previous pre-diabetes", "Presence of hypertension","Systolic blood pressure",
        "Diastolic blood pressure","Fasting blood sugar levels", "Hemoglobin A1c levels", "Serum creatinine levels", "Blood Urea Nitrogen levels",
        "Total cholesterol levels", "Low-density lipoprotein cholesterol levels", "High-density lipoprotein cholesterol levels","Triglycerides levels",
        "Use of antihypertensive medications", "Use of statins","Use of antidiabetic medications", "Presence of frequent urination", "Presence of excessive thirst",
        "Presence of unexplained weight loss","Fatigue levels", "Presence of blurred vision", "Presence of slow-healing sores",
        "Presence of tingling in hands or feet", "Quality of life score", "Exposure to heavy metals", "Occupational exposure to harmful chemicals", "Quality of water",
        "Frequency of medical check-ups per year", "Medication adherence score", "Health literacy score","Diagnosis status for Diabetes"
    ],
    "Data Type": [
        "Numeric", "Binary", "Nominal", "Ordinal", "Ordinal",
        "Numeric", "Binary", "Numeric", "Numeric", "Numeric",
        "Numeric", "Binary", "Binary", "Binary", "Binary", "Binary",
        "Numeric", "Numeric", "Numeric", "Numeric",
        "Numeric", "Numeric", "Numeric", "Numeric",
        "Numeric", "Numeric", "Binary", "Binary", "Binary",
        "Binary", "Binary", "Binary", "Numeric", "Binary", "Binary",
        "Binary", "Numeric", "Binary", "Binary", "Binary",
        "Numeric", "Numeric", "Numeric", "Binary"
    ],
    "Possible Values": [
        "20-90", "0(Male), 1(Female)", "0(Caucasian), 1(African American), 2(Asian), 3(Other)",
        "0(Low), 1(Middle), 2(High)", "0(None), 1(High School), 2(Bachelor's), 3(Higher)", "15-40",
        "0(No), 1(Yes)", "0-20","0-10", "0-10", "4-10", "0(No), 1(Yes)", "0(No), 1(Yes)", "0(No), 1(Yes)",
        "0(No), 1(Yes)", "0(No), 1(Yes)", "90-180 mmHg", "60 -120 mmHg","70-200 mg/dL", "4.0%-10.0%",
        "0.5-5.0 mg/dL", "5-50 mg/dL", "150-300 mg/dL", "50-200 mg/dL", "20-100 mg/dL", "50-400 mg/dL",
        "0(No), 1(Yes)","0(No), 1(Yes)", "0(No), 1(Yes)", "0(No), 1(Yes)", "0(No), 1(Yes)", "0(No), 1(Yes)",
        "0-10","0(No), 1(Yes)","0(No), 1(Yes)", "0(No), 1(Yes)","0-100", "0(No), 1(Yes)","0(No), 1(Yes)",
        "0(Good), 1(Poor)", "0-4", "0-10", "0-10","0(No), 1(Yes)"
    ]
}

attribute_df = pd.DataFrame(data)
attribute_df

Attribute Name  \
0                             Age   
1                          Gender   
2                       Ethnicity   
3             SocioeconomicStatus   
4                  EducationLevel   
5                             BMI   
6                         Smoking   
7              AlcoholConsumption   
8                PhysicalActivity   
9                     DietQuality   
10                   SleepQuality   
11          FamilyHistoryDiabetes   
12            GestationalDiabetes   
13        PolycysticOvarySyndrome   
14            PreviousPreDiabetes   
15                   Hypertension   
16                     SystolicBP   
17                    DiastolicBP   
18              FastingBloodSugar   
19                          HbA1c   
20                SerumCreatinine   
21                      BUNLevels   
22               CholesterolTotal   
23                 CholesterolLDL   
24                 CholesterolHDL   
25       CholesterolTriglycerides   
26    AntihypertensiveMedications   
27                        Statins   
28        AntidiabeticMedications   
29              FrequentUrination   
30                ExcessiveThirst   
31          UnexplainedWeightLoss   
32                  FatigueLevels   
33                  BlurredVision   
34               SlowHealingSores   
35              TinglingHandsFeet   
36             QualityOfLifeScore   
37            HeavyMetalsExposure   
38  OccupationalExposureChemicals   
39                   WaterQuality   
40       MedicalCheckupsFrequency   
41            MedicationAdherence   
42                 HealthLiteracy   
43                      Diagnosis   

                                    Description Data Type  \
0                       The age of the patients   Numeric   
1                        Gender of the patients    Binary   
2                 The ethnicity of the patients   Nominal   
3      The socioeconomic status of the patients   Ordinal   
4           The education level of the patients   Ordinal   
5               Body Mass Index of the patients   Numeric   
6                                Smoking status    Binary   
7           Weekly alcohol consumption in units   Numeric   
8             Weekly physical activity in hours   Numeric   
9                            Diet quality score   Numeric   
10                          Sleep quality score   Numeric   
11                   Family history of diabetes    Binary   
12              History of gestational diabetes    Binary   
13        Presence of polycystic ovary syndrome    Binary   
14             History of previous pre-diabetes    Binary   
15                     Presence of hypertension    Binary   
16                      Systolic blood pressure   Numeric   
17                     Diastolic blood pressure   Numeric   
18                   Fasting blood sugar levels   Numeric   
19                        Hemoglobin A1c levels   Numeric   
20                      Serum creatinine levels   Numeric   
21                   Blood Urea Nitrogen levels   Numeric   
22                     Total cholesterol levels   Numeric   
23   Low-density lipoprotein cholesterol levels   Numeric   
24  High-density lipoprotein cholesterol levels   Numeric   
25                         Triglycerides levels   Numeric   
26          Use of antihypertensive medications    Binary   
27                               Use of statins    Binary   
28              Use of antidiabetic medications    Binary   
29               Presence of frequent urination    Binary   
30                 Presence of excessive thirst    Binary   
31          Presence of unexplained weight loss    Binary   
32                               Fatigue levels   Numeric   
33                   Presence of blurred vision    Binary   
34               Presence of slow-healing sores    Binary   
35        Presence of tingling in hands or feet    Binary   
36                        Quality of life score   Numeric   
37                     Exposure to 

# Data Summary:

In [ ]:
# Extracting the columns of 'Numeric' data type and arranging them in a list
numeric_columns = attribute_df[attribute_df['Data Type'] == 'Numeric']['Attribute Name'].tolist()

# Create a new data frame with only 'Numeric' columns
numeric_df = df[numeric_columns]

# Show all columns without truncation
pd.set_option('display.max_columns', None)
summary = numeric_df.describe()
summary

Age          BMI  AlcoholConsumption  PhysicalActivity  \
count  1879.000000  1879.000000         1879.000000       1879.000000   
mean     55.043108    27.687601           10.096587          5.200790   
std      20.515839     7.190975            5.914216          2.857012   
min      20.000000    15.025898            0.000928          0.004089   
25%      38.000000    21.469981            4.789725          2.751022   
50%      55.000000    27.722988           10.173865          5.249002   
75%      73.000000    33.856460           15.285359          7.671402   
max      90.000000    39.998811           19.996231          9.993893   

       DietQuality  SleepQuality   SystolicBP  DiastolicBP  FastingBloodSugar  \
count  1879.000000   1879.000000  1879.000000  1879.000000        1879.000000   
mean      4.895801      7.021328   134.050559    89.863757         135.204490   
std       2.867144      1.729469    25.613830    17.328086          37.515750   
min       0.000885      4.004336    90.000000    60.000000          70.074649   
25%       2.476802      5.481789   112.000000    75.000000         102.341470   
50%       4.888566      7.094692   134.000000    90.000000         137.398241   
75%       7.356058      8.523814   156.000000   105.000000         167.362362   
max       9.998677      9.989372   179.000000   119.000000         199.935506   

             HbA1c  SerumCreatinine    BUNLevels  CholesterolTotal  \
count  1879.000000      1879.000000  1879.000000       1879.000000   
mean      6.976133         2.784590    27.798153        225.006464   
std       1.739365         1.308023    12.800797         43.367170   
min       4.003089         0.500565     5.010401        150.056094   
25%       5.443856         1.654472    17.172009        186.933051   
50%       7.095732         2.855105    28.190147        225.120112   
75%       8.423211         3.898133    38.506299        263.529424   
max       9.991193         4.993974    49.975728        299.998480   

       CholesterolLDL  CholesterolHDL  CholesterolTriglycerides  \
count     1879.000000     1879.000000               1879.000000   
mean       124.656831       60.060944                227.386167   
std         42.911145       23.316682                101.071578   
min         50.058252       20.014494                 50.154649   
25%         87.810946       40.011963                140.873930   
50%        124.918023       60.456988                228.417429   
75%        161.624444       80.059112                313.408944   
max        199.898732       99.958394                399.885928   

       FatigueLevels  QualityOfLifeScore  MedicalCheckupsFrequency  \
count    1879.000000         1879.000000               1879.000000   
mean        4.949003           48.508643                  1.997101   
std         2.884483           28.758488                  1.122632   
min         0.004977            0.002390                  0.004013   
25%         2.417748           23.974098                  1.057801   
50%         4.851914           47.519693                  1.987170   
75%         7.569772           72.883179                  2.946019   
max         9.999979           99.788530                  3.999715   

       MedicationAdherence  HealthLiteracy  
count          1879.000000     1879.000000  
mean              4.957539        5.011736  
std               2.910934        2.920908  
min               0.005384        0.000362  
25%               2.420024        2.410113  
50%               4.843886        5.035208  
75%               7.513933        7.586865  
max               9.997165        9.993029

# Statistical Summary Of Numeric Attribute:

* **Age**  
  
The ages of patients range from 20 to 90 years, with an average of 55 years. The median age is 55 years suggesting a relatively balanced age distribution. The standard deviation is 20.52 indicating moderate to high variability, meaning the dataset contains both young and older individuals.  

* **BMI**  

Body Mass Index ranges from 15 to 39.9, with a mean of 27.69 and a median of 27.72 indicateing mostly overweight. The standard deviation is 7.19 showing a wide spread of BMI values, reflecting a mix of underweight, normal, overweight, and obese individuals.  

* **AlcoholConsumption**  

Weekly alcohol consumption ranges from 0.0009 to 19.99 units, with an average of 10.10 units. The median is 10.17 indicating a fairly symmetric distribution. Standard deviation is 5.91 indicating moderate variability where most people consume roughly 4 to 16 units per week.

* **PhysicalActivity**  

Weekly physical activity ranges from 0.004 to 9.99 hours, with a mean of 5.20 and a median of 5.25 hours. The standard deviation is 2.86 indicating moderate variability among patients’ activity levels where most people engage in roughly 2 to 8 hours per week.

* **DietQuality**  

Diet quality score ranges from 0.0009 to 9.99, with an average of 4.90 and median of 4.89. Standard deviation is 2.87 suggesting that scores are spread out but still mostly clustered around the middle among patients.

* **SleepQuality**  

Sleep quality ranges from 4.00 to 9.99, with a mean of 7.02 and a median of 7.09. Standard deviation is 1.73 indicating low to moderate variability in sleep quality.

* **SystolicBP**  

Values range from 90 to 179 mmHg, with a mean of 134.05 and median of 134 mmHg. Standard deviation is 25.61, indicating high variability in systolic blood pressure with values spread widely from normal to high levels.

* **DiastolicBP**  

Diastolic BP ranges from 60 to 119 mmHg, with mean 89.86 and median 90 mmHg. Standard deviation is 17.33 reflecting relatively high variability in diastolic blood pressure, with values spread from normal to hypertensive levels.

* **FastingBloodSugar**  

Values range from 70.07 to 199.94 mg/dL, mean is 135.20, and median 137.40. Standard deviation is 37.52, showing significant variability, indicating diverse glucose levels among patients, with results extending from normal to elevated levels.

* **HbA1c**  

Ranges from 4.00% to 9.99%, mean 6.98%, median 7.10%, standard deviation 1.74%, suggesting moderate variability with most values concentrated near the mean but still showing some spread in long-term blood glucose control.
 
* **SerumCreatinine**  

Ranges from 0.5 to 5.0 mg/dL, with a mean of 2.78. The median is 2.85, close to the mean. The standard deviation of 1.31 shows moderate variation, with most values lying between roughly 1.5 and 4.0 mg/dL.

* **BUNLevels**  

Range from 5.0 to 50 mg/dL, with a mean of 27.8. The median of 28.2 is close to the mean, indicating balanced distribution. The standard deviation of 12.8 shows moderate to high variability, with values extending from low to high in blood urea nitrogen levels.

* **CholesterolTotal**  

Values range from 150 to 300 mg/dL, with a mean of 225. The median is 225.1, almost identical. The standard deviation of 43.4 indicates significant variation, which may reflect different cardiovascular risk profiles.

* **CholesterolLDL**  

Ranges from 50 to 200 mg/dL, with a mean of 125 and a median of 125, almost identical. The standard deviation of 42.9 indicates a wide spread, with individuals having both low and very high LDL levels.

* **CholesterolHDL**  

Ranges from 20 to 100 mg/dL, with a mean of 60.1. The median of 60.5 is close to the mean, suggesting balanced distribution. The standard deviation of 23.3 showing relatively high variability, with values extending from low to high levels.

* **CholesterolTriglycerides**  

Values range from 50 to 400 mg/dL, with a mean of 227.4 The median is 228.4, close to the mean. The standard deviation of 101 shows substantial variability, suggesting many individuals with elevated triglyceride levels.

* **FatigueLevels**  

Scores range from near 0 to 10, with a mean of 4.95. The median of 4.85 is close to the mean. The standard deviation of 2.88 suggests moderate variability in reported fatigue.

* **QualityOfLifeScore**  

Ranges from 0 to 100, with a mean of 48.5. The median is 47.5, showing balance. The standard deviation of 28.8 indicates wide differences in perceived quality of life among individuals.

* **MedicalCheckupsFrequency**  

Ranges from 0 to 4, with a mean of 2.0. The median of 1.99 is almost identical, indicating a balanced distribution. The standard deviation of 1.12 shows mild variability in checkup habits.

* **MedicationAdherence**  

Ranges from near 0 to 10, with a mean of 4.96. The median of 4.84 is close to the mean, showing balance. The standard deviation of 2.91 shows moderate variation, with most participants near the average but some with higher or lower scores in adherence levels.

* **HealthLiteracy**  

Ranges from near 0 to 10, with a mean of 5.01. The median of 5.04 is very close to the mean, suggesting symmetry. The standard deviation of 2.92 shows moderate variability in health literacy levels.

# Missing Values Analysis:

* (False) means there are no missing values.  
* (True) means there are missing values.

In [8]:
# Checkign all data frame for any missing values
df.isnull().values.any()

np.False_